RELEASE_PROCESS.md

```diff
@@ -0,0 +1,44 @@
+# TektonCD CLI release process
+
+- You need to be a tektoncd-cli [OWNERS](OWNERS) to be able to push the new branch
+
+- You need to have your own tekton cluster, under `minikube`, `gcp` or other means.
```

Comment:
nit: at some point, we want to automate release on the `dogfooding` cluster :wink: but we are not yet there for sure :sweat_smile:

Reply:
:)